### ISSUE TITLE
Rule "RequireTrailingCommaInDeclaration" is interfering other rules

### ISSUE THREAD

**Issue by denis-rossati** [slevomat/coding-standard#1335]
The rule `SlevomatCodingStandard.Functions.RequireTrailingCommaInDeclaration` is interfering other rules and messing the expected correction.

Given the following code to be corrected:

<details>

<Summary>Code to be corrected</Summary>

```php
<?php

declare(strict_types=1);

static function ():void
{
}

static function (): void
{
}

static function ():   void
{
}

static function () :void
{
}

static function ()   :void
{
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
):void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
) :void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
)   :void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
):   void {
}
```
</details>

when the `phpcbf` is executed, the expected correction should be

<details>

<Summary>Corrected code</Summary>

```php
<?php

declare(strict_types=1);

static function (): void {
}

static function (): void {
}

static function (): void {
}

static function (): void {
}

static function (): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}
```
</details>

but it does not happen and the incorrect code remains untouched, when this rule is in the code standard document, this correction is simply ignored. When I remove the rule, everything works fine and the code is corrected correctly.

**Comment by kukulich** [comment]
@denis-rossati 

> the code to be corrected is not is not modified. But when I remove that line, the code is corrected as expected

I'm sorry I still don't undestand this.

**Comment by denis-rossati** [comment]
That's okay 😅 maybe I'm not explaining too well.

When I run the `phpcbf`, I expected this snippet 

<details>
<Summary>Code with layout issues</Summary>

```php

<?php

declare(strict_types=1);

static function ():void
{
}

static function (): void
{
}

static function ():   void
{
}

static function () :void
{
}

static function ()   :void
{
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
):void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
) :void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
)   :void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
):   void {
}
```
</details>

to be corrected to look like this snippet

<details>
<Summary>Code corrected</Summary>

```php

<?php

declare(strict_types=1);

static function (): void {
}

static function (): void {
}

static function (): void {
}

static function (): void {
}

static function (): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}
```
</details>

I have other layout rules that works well, but when I add the rule "SlevomatCodingStandard.Functions.RequireTrailingCommaInDeclaration" the code remais like this

<details>
<Summary>Code with layout issues</Summary>

```php

<?php

declare(strict_types=1);

static function ():void
{
}

static function (): void
{
}

static function ():   void
{
}

static function () :void
{
}

static function ()   :void
{
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
):void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
) :void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
)   :void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
): void {
}

static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
):   void {
}
```
</details>

**Comment by kukulich** [comment]
@denis-rossati `RequireTrailingCommaInDeclaration` should fix this:

```
static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b
):void {
}
```

to this:

```
static function (
    int $a,
    int $c,
    int $d,
    int $e,
    int $b,
):void {
}
```

This sniff does not anything else.

**Comment by denis-rossati** [comment]
@kukulich I see, but when `RequireTrailingCommaInDeclaration` is added, it stop other rules work

**Comment by denis-rossati** [comment]
I see. But when this rule is added, looks like it interferes in other rules. Here is a fragment of the test report when the rule is included:

```sh
3. Fix tests
✗ Fix tests failed!

Result:
    diff tests/Integration/tmp/return-type-spacing-on-closures.php tests/Integration/fixed/return-type-spacing-on-closures.php
    5,6c5
    < static function ():void
    < {
    ---
    > static function (): void {
    9,10c8
    < static function (): void
    < {
    ---
    > static function (): void {
    13,14c11
    < static function ():   void
    < {
    ---
    > static function (): void {
    17,18c14
    < static function () :void
    < {
    ---
    > static function (): void {
    21,22c17
    < static function ()   :void
    < {
    ---
    > static function (): void {
    31c26
    < ):void {
    ---
    > ): void {
    40c35
    < ) :void {
    ---
    > ): void {
    49c44
    < )   :void {
    ---
    > ): void {
    67c62
    < ):   void {
    ---
    > ): void {
```

Even though the rule shouldn't do nothing related to this particular code diff, appears that the rule is responsible for this bug because when removed, the test passes without problems.

**Comment by kukulich** [comment]
What’s is your PHPCS configuration?

**Comment by denis-rossati** [comment]
Is [right here](https://github.com/denis-rossati/Croct-code-standard), I'm sorry for the wait

**Comment by kukulich** [comment]
@denis-rossati I'm not able to reproduce it. Please try to write failing test.

**Comment by denis-rossati** [comment]
Great day! Turns out that I was using `DisallowTrailingCommaInDeclaration` instead of `RequireTrailingCommaInDeclaration` 🤦

I'm very sorry for the erroneous issue. Thanks for your time and patience!
